geomesa-fs/geomesa-fs-datastore/src/main/scala/org/locationtech/geomesa/fs/FileSystemDataStoreFactory.scala

```diff
@@ -37,7 +37,8 @@ class FileSystemDataStoreFactory extends DataStoreFactorySpi {
       props.asScala.foreach { case (k, v) => conf.set(k, v) }
       conf
     }.getOrElse(configuration)
-
+    CharsetParam.lookupOpt(params)
+      .map(v => conf.set(CharsetParam.key,v))
```

Comment:
the `conf` here could be the cached singleton instance from the `FileSystemDataStoreFactory` object, so we don't want to modify it directly.

Reply:
Thank you for your review. I should learn more about the coding style which accepted by the majority . About the charset configuration ,I agree with you ,though I change  some lines to make it look better .  I will improve myself so that I can talk with you at the same level . Thank you for your time ! And I wish to communicate with you  more !

Reply:
no problem, thanks for contributing!